system/system.go

```diff
@@ -151,6 +153,101 @@ func (p *PingOperation) Execute(ctx context.Context, c *internal.Clients) ([]*sp
 	}
 }
 
+// RebootOperation represents the parameters of a Reboot operation.
+type RebootOperation struct {
+	rebootMethod         spb.RebootMethod
+	delay                time.Duration
+	message              string
+	subcomponents        []*tpb.Path
+	force                bool
+	ignoreUnavailableErr bool
+	waitForActive        bool
+}
+
+// NewRebootOperation creates an empty RebootOperation.
+func NewRebootOperation() *RebootOperation {
+	return &RebootOperation{}
+}
+
+// RebootMethod specifies method to reboot.
+func (r *RebootOperation) RebootMethod(rebootMethod spb.RebootMethod) *RebootOperation {
+	r.rebootMethod = rebootMethod
+	return r
+}
+
+// Delay specifies time in nanoseconds to wait before issuing reboot.
+func (r *RebootOperation) Delay(delay time.Duration) *RebootOperation {
+	r.delay = delay
+	return r
+}
+
+// Message specifies informational reason for the reboot or cancel reboot.
+func (r *RebootOperation) Message(message string) *RebootOperation {
+	r.message = message
+	return r
+}
+
+// Subcomponents specifies the sub-components to reboot.
+func (r *RebootOperation) Subcomponents(subcomponents []*tpb.Path) *RebootOperation {
+	r.subcomponents = subcomponents
+	return r
+}
+
+// Force reboot if sanity checks fail.
+func (r *RebootOperation) Force(force bool) *RebootOperation {
+	r.force = force
+	return r
+}
+
+// IgnoreUnavailableErr ignores unavailable errors returned by reboot status.
+func (r *RebootOperation) IgnoreUnavailableErr(ignoreUnavailableErr bool) *RebootOperation {
```

Comment:
I think we just drop this and always ignore UnavailableErrors when waiting for active

Reply:
hmmm.. is UnavailableError always acceptable after reboot? 

Having a function to ignore the error gives the flexibility to ignore errors based on the usecase. 

I do see a test here which performs a reboot and expects error to be nil -> https://github.com/openconfig/featureprofiles/blob/main/feature/gnoi/system/tests/chassis_reboot_status_and_cancel_test/chassis_reboot_status_and_cancel_test.go#L168

Reply:
Yeah, I see the pros and cons. Three comments:

First, It just seems to me that the likelihood that we would see an unexpected Unavailable error it is very low, especially given that the prior RebootRequest must have succeeded by then. Other errors would obviously still be reported.

Second, almost all uses of `WaitForActive(true)` is going to want to be accompanied by `IgnoreUnavailableErr(true)`, because nearly all will be rebooting the chassis, and if not that the management linecard. So I feel like `IgnoreUnavailableErr(true)` is going to be one of these lines people just copy and paste everywhere anyway, regardless of whether they need it, because that's what most calls look like.

Third, I'm holding out some hope that we can extend the `RebootRequestResponse` to make `IgnoreUnavailableErr` eventually unnecessary. I'm going to be starting a thread on the OpenConfig chat about that.

Due to those three, I was feeling like IgnoreUnavailableErr maybe isn't offering enough to pull its weight. All that said, I still see the reasons for including it, so I'm fine with you leaving it in if you prefer to keep it.